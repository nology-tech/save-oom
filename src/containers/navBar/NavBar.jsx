import React from "react";
import "./NavBar.scss";
import NavBarHeading from "../../components/navBarHeading/NavBarHeading";
import vector from "../../assets/images/Vectorvector.png";
import Button from "../../components/Button/Button";
import Logo from "../../components/Logo/Logo";
import lockImg from "../../assets/images/Vectorlock.png";
import statImg from "../../assets/images/Vectorstat.png";
import addImg from "../../assets/images/Vectoradd.png";

const NavBar = () => {
  return (
    <div className="nav-bar">
      <div className="nav-bar__section">
        <Logo />
        <NavBarHeading
          headingText={"Phonics stages"}
          headingImg={vector}
          headingStyle={"nav-bar"}
        />
        <h2>Introduction</h2>
        <NavBarHeading
          headingText={"Level 1"}
          headingImg={lockImg}
          headingStyle={"nav-bar"}
        />
        <NavBarHeading
          headingText={"Level 2"}
          headingImg={lockImg}
          headingStyle={"nav-bar"}
        />
        <NavBarHeading
          headingText={"Level 3"}
          headingImg={lockImg}
          headingStyle={"nav-bar"}
        />
        <NavBarHeading
          headingText={"Stats"}
          headingImg={statImg}
          headingStyle={"nav-bar"}
        />
        <h2>Jack</h2>
        <NavBarHeading
          headingText={"Add Avatar"}
          headingImg={addImg}
          headingStyle={"nav-bar"}
        />
      </div>
      <Button buttonText={"Logout"} buttonStyle={"button-logout"} />
    </div>
  );
};

export default NavBar;
