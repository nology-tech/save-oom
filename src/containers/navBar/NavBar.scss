@use "../../styles/reset" as *;
@use "../../styles/variables" as *;
@use "../../styles/typography" as *;

.nav-bar {
  background-color: $color-secondary-dark;
  padding: 1rem;
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  max-width: 20rem;
}
