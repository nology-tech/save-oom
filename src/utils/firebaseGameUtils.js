import { db } from "../firebase";
import {
  addDoc,
  collection,
  getDocs,
  query,
  where,
  orderBy,
  Timestamp,
} from "firebase/firestore";
/** 
 * Get all the rounds of a specific game and level for a given user, ordered by most recent first 
 @param {*} userId
  @param {String} game
*@returns a querySnap (document snapshot), returns a Promise.
*/

export const getGameRoundsForUser = async (id, game) => {
  const q = query(
    collection(db, "users", id, "rounds_played"),
    where("gameId", "==", game),
    orderBy("roundTime", "desc")
  );
  const querySnap = await getDocs(q);
  return querySnap;
};

/** 
   * Get all the correct round of a specific game and level for a given user, ordered by most recent first 
  @param {String} userId
  @param {String} gameId
  @param {Number} level  
  *@returns a promise with the query snapshot.
  */
export const getCorrectGameRoundsForUser = async (id, game, level) => {
  const q = query(
    collection(db, "users", id, "rounds_played"),
    where("gameId", "==", game),
    where("level", "==", level),
    where("correct", "==", true),
    orderBy("roundTime", "desc")
  );
  const querySnap = await getDocs(q);
  return querySnap;
};

/** Get all the incorrect rounds for a given user, ordered by most recent first 
  @param {String} userId
  *@returns  a promise with the query snapshot.
  */
export const getIncorrectGameRoundsForUser = async (id) => {
  const q = query(
    collection(db, "users", id, "rounds_played"),
    where("correct", "==", false),
    orderBy("roundTime", "desc")
  );
  const querySnap = await getDocs(q);
  return querySnap;
};

/** Get all the incorrect round of a specific game for a given user, ordered by most recent first 
  @param {String} userId
  @param {String} gameId
  @param {Number} level  
  *@returns  a promise with the query snapshot.
  */

export const getAllIncorrectGameRoundsForUser = async (id, game) => {
  const q = query(
    collection(db, "users", id, "rounds_played"),
    where("gameId", "==", game),
    where("correct", "==", true),
    orderBy("roundTime", "desc")
  );
  const querySnap = await getDocs(q);
  return querySnap;
};



/** Function that returns an array of rounds 
  @param {String} userId
  @param {String} gameId
  @param {Function} getGameRounds
  *@returns returs a promise.
  */
export const getArrayOfRounds = async (id, game, level, getGameRounds) => {
  const querySnap = await getGameRounds(id, game, level);
  const phonicsArr = querySnap.docs.map((doc) => {
    return doc.data().phonic;
  });
  return phonicsArr;
};

/** Function that returns an array of rounds 
  @param {String} id
  @param {String} game
  @param {Object} roundInfo
  @param {String} answer
  *@returns returs a promise.
  */

export const saveUserRound = async (id, game, level, gameStats, answer) => {
  const { isCorrect, score } = gameStats;
  await addDoc(collection(db, "users", id, "rounds_played"), {
    correct: isCorrect,
    gameId: game,
    phonic: answer,
    level: level,
    roundTime: Timestamp.fromDate(new Date()).toDate(),
    score: score,
  });
};
