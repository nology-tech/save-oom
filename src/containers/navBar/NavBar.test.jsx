import "./NavBar";
import NavBar from "./NavBar";
import UserContext from "../../contexts/UserContext";
import {customRender, customContextRender} from "../../utils/testUtils"
import { screen } from "@testing-library/react";

const userObj = {
  user: {
    name: "Jack", 
    email:"Hello@gmail"
  }
};

it("Should render the navbar", () => {
  const { container } = customRender(
    <UserContext.Provider value={userObj}>
      <NavBar />
    </UserContext.Provider>
  );
  expect(container).toMatchSnapshot();
});

it("Should render the navbar", () => {
  const { container } = customRender(
    <UserContext.Provider value={userObj}>
      <NavBar />
    </UserContext.Provider>
  );
  expect(container).toMatchSnapshot();
});


it("Should contain the navbar list that contains levels, stats and name from context provider ", () => {
  customContextRender(<NavBar />, userObj);
  const navItem1 = screen.getByText(/Level 1/i);
  const navItem2 = screen.getByText(/Level 2/i);
  const name = screen.getByText(/Jack/i);
  const stats = screen.getByText(/Stats/i);
  expect(navItem1).toBeInTheDocument();
  expect(navItem2).toBeInTheDocument();
  expect(stats).toBeInTheDocument();
  expect(name).toBeInTheDocument();
});
