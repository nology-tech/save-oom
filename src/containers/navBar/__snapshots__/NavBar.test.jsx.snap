// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render the navbar 1`] = `
<div>
  <div
    class="nav-bar"
  >
    <div
      class="nav-bar__section"
    >
      <img
        alt="Logo image"
        class="logo__image"
        src="logo.png"
      />
      <div
        class="nav-bar-header"
      >
        <h2
          class="nav-bar-header__heading"
        >
          Phonics stages
        </h2>
      </div>
      <h2>
        Introduction
      </h2>
      <div
        class="nav-bar-header"
      >
        <h2
          class="nav-bar-header__heading"
        >
          Level 1
        </h2>
      </div>
      <div
        class="nav-bar-header"
      >
        <h2
          class="nav-bar-header__heading"
        >
          Level 2
        </h2>
      </div>
      <div
        class="nav-bar-header"
      >
        <h2
          class="nav-bar-header__heading"
        >
          Level 3
        </h2>
      </div>
      <div
        class="nav-bar-header"
      >
        <h2
          class="nav-bar-header__heading"
        >
          Stats
        </h2>
      </div>
      <h2>
        Jack
      </h2>
      <div
        class="nav-bar-header"
      >
        <h2
          class="nav-bar-header__heading"
        >
          Add Avatar
        </h2>
      </div>
    </div>
    <button
      class="button-logout"
    >
      Logout
    </button>
  </div>
</div>
`;

exports[`Should render the navbar 2`] = `
<div>
  <div
    class="nav-bar"
  >
    <div
      class="nav-bar__section"
    >
      <img
        alt="Logo image"
        class="logo__image"
        src="logo.png"
      />
      <div
        class="nav-bar-header"
      >
        <h2
          class="nav-bar-header__heading"
        >
          Phonics stages
        </h2>
      </div>
      <h2>
        Introduction
      </h2>
      <div
        class="nav-bar-header"
      >
        <h2
          class="nav-bar-header__heading"
        >
          Level 1
        </h2>
      </div>
      <div
        class="nav-bar-header"
      >
        <h2
          class="nav-bar-header__heading"
        >
          Level 2
        </h2>
      </div>
      <div
        class="nav-bar-header"
      >
        <h2
          class="nav-bar-header__heading"
        >
          Level 3
        </h2>
      </div>
      <div
        class="nav-bar-header"
      >
        <h2
          class="nav-bar-header__heading"
        >
          Stats
        </h2>
      </div>
      <h2>
        Jack
      </h2>
      <div
        class="nav-bar-header"
      >
        <h2
          class="nav-bar-header__heading"
        >
          Add Avatar
        </h2>
      </div>
    </div>
    <button
      class="button-logout"
    >
      Logout
    </button>
  </div>
</div>
`;
